src/pybind/mgr/volumes/fs/async_cloner.py

```diff
@@ -170,12 +177,17 @@ def cptree(src_root_path, dst_root_path):
                         elif stat.S_ISREG(stx["mode"]):
                             log.debug("cptree: (REG) {0}".format(d_full_src))
                             copy_file(fs_handle, d_full_src, d_full_dst, mo, cancel_check=should_cancel)
+                            stats.size_h += stx['size']
```

Comment:
> What's the need to track the amount of data that's copied? 

Progress bar won't work otherwise. Also, we need to print percentage completed in `fs clone status` command's output.

> This can be fetched via rstats

That's not possible. Actual size copied will reflect in rstats after more than a couple of seconds; in such case progress bar won't work at all. Getting size from stat buffer is also faster and easier than running `getfattr` command and then parsing its output. Since we already have stat buffer here, it's even better.

Reply:
> > What's the need to track the amount of data that's copied?
> 
> Progress bar won't work otherwise. Also, we need to print percentage completed in `fs clone status` command's output.
> 
> > This can be fetched via rstats
> 
> That's not possible. Actual size copied will reflect in rstats after more than a couple of seconds; in such case progress bar won't work at all.

If a clone operations finishes under 2s, it's totally fine to have no progress bar at all, isn't it? Also, once a clone finishes (for the under 2s clone, say), recalculating clone progress would take into account other on-going+pending clones.

> Getting size from stat buffer is also faster and easier than running `getfattr` command and then parsing its output. Since we already have stat buffer here, it's even better.

The client would have cached the update (very likely).

So, my question is - where exactly do you plan to use rstats then if explicit tracking of stats is done during clone?